tardis/model/base.py

```diff
@@ -376,16 +393,43 @@ def density(self):
         return self.model_state.density
 
     @property
-    def abundance(self):
-        if not self.raw_isotope_abundance.empty:
-            self._abundance = self.raw_isotope_abundance.decay(
-                self.time_explosion
-            ).merge(self.raw_abundance)
-        abundance = self._abundance.loc[
-            :, self.v_boundary_inner_index : self.v_boundary_outer_index - 1
+    def isotope_abundance(self):
+        raw_abundance = self.raw_abundance.copy()
+        raw_abundance["mass_number"] = -1
+        raw_abundance = raw_abundance.set_index(
+            [raw_abundance.index, "mass_number"]
+        )
+
+        raw_abundance.columns = pd.RangeIndex(start=0, stop=len(raw_abundance.columns), step=1)
+
+        if not self._raw_isotope_abundance.empty:
+            self._abundance = raw_abundance.append(
+                self._raw_isotope_abundance.decay(self.time_explosion)
+            )
+            norm_factor = self._abundance.sum(axis=0)
+            self._abundance /= norm_factor
+        else:
+            self._abundance = raw_abundance
+
+        isotope_abundance = self._abundance.iloc[
+            :, self.v_boundary_inner_index : self.v_boundary_outer_index
         ]
-        abundance.columns = range(len(abundance.columns))
-        return abundance
+
+        isotope_abundance.columns = range(len(isotope_abundance.columns))
+        return isotope_abundance
+
+    @property
+    def raw_isotope_abundance(self):
+        return self._raw_isotope_abundance
+
+    @raw_isotope_abundance.setter
+    def raw_isotope_abundance(self, value):
+        self._raw_isotope_abundance = value
+        self.model_state.isotope_abundance = self.isotope_abundance
```

Comment:
In [this](https://github.com/tardis-sn/tardis/blob/c70594674b4e0998f3ea8a349911c1096e877331/tardis/model/tests/test_base.py#L283) test, `raw_isotope_abundance` is redefined. The test used to fail because the change was not reflected in `model_state`'s `isotope_abundance` and correspondingly in the `abundance` property.

I thought defining a setter method would be a good idea to make sure that `model_state`'s `isotope_abundance` is updated as well.